pkg/config/validation/validation.go

```diff
@@ -2054,8 +2065,26 @@ var ValidateAuthorizationPolicy = registerValidateFunc("ValidateAuthorizationPol
 					}
 				}
 			}
+
+			if !tcpRulesInFrom && !tcpRulesInTo && ((rule.To != nil && len(rule.To) != 0) ||
+				(rule.From != nil && len(rule.From) != 0)) && in.Action == security_beta.AuthorizationPolicy_DENY {
+				ns := cfg.Meta.Namespace
+				if in.Selector != nil && in.Selector.MatchLabels != nil {
+					var label string
+					for key, val := range in.Selector.MatchLabels {
+						label = label + " " + key + ":" + val
+					}
+					warning = fmt.Errorf("all the traffic to TCP ports for pods with labels %s in"+
+						"namespace %s will be denied because your deny policy has a pure L7 rule,"+
+						"it's better to add ports when you define any deny rule ", ns, label)
```

Comment:
Its a bit confusing that we are looking for "Pure L7". That means this doesn't trigger it:

```
spec:
  action: DENY
  rules:
  - 
    from:
    - source:
        principals: ["cluster.local/ns/default/sa/sleep"]
    to:
    - operation:
        methods: ["GET"]
```

Users intent: "Deny GET from sleep"
Actual impact: "Deny everything from sleep"

This PR would *not* warn about that at all.

Reply:
The attached policy has info for TCP filter as well (`Principals`). The behavior would be, request will be denied if it a GET req or coming from any peer with identity `cluster.local/ns/default/sa/sleep`. IMO it is a valid policy and we shouldn't warn about it.